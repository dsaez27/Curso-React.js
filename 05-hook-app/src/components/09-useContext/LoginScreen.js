import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginScreen = () => {
    const { setUser } = useContext(UserContext);

    const handleLogin = () => {
        setUser({
            id: 123,
            name: 'Daniel',
            email: 'dssh27@gmail.com'
        })
    }

    return (
        <>
            <h1>LoginScreen</h1>
            <hr />
            <button className='btn btn-primary' onClick={handleLogin}>
                Login
            </button>
        </>
    );
};
