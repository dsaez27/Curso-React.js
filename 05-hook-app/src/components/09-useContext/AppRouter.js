import React from 'react';
import { BrowserRouter as Router, Switch, Route, Redirect } from 'react-router-dom';
import { LoginScreen } from './LoginScreen';
import { HomeScreen } from './HomeScreen';
import { AboutScreen } from './AboutScreen';
import { NavBar } from './NavBar';

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar />
                <Switch>
                    <Route path="/" exact component={HomeScreen} />
                    <Route path="/about" exact component={AboutScreen} />
                    <Route path="/login" exact component={LoginScreen} />
                    <Redirect to="/" />
                </Switch>
            </div>
        </Router>
    )
}
