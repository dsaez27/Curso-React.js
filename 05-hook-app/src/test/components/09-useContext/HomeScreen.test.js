import React from 'react'
import { HomeScreen } from './../../../components/09-useContext/HomeScreen';
import { mount } from 'enzyme';
import { UserContext } from './../../../components/09-useContext/UserContext';

describe('Pruebas en <HomeScreen />', () => {

    const user = {
        name: 'daniel',
        email: 'daniel@example.com',
    }
    
    const wrapper = mount(
        <UserContext.Provider value={{
            user
        }} >
            <HomeScreen />
        </UserContext.Provider>);
    test('Debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
})
