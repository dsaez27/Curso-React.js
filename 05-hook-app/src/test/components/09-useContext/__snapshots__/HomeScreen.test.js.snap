// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeScreen /> Debe de mostrarse correctamente 1`] = `
Array [
  <h1>
    HomeScreen
  </h1>,
  <hr />,
  <pre>
    {
   "name": "daniel",
   "email": "daniel@example.com"
}
  </pre>,
]
`;
